### PR TITLE
CB-12016 : removes pluginId mapping

### PR DESCRIPTION
**Pull request by audreyso**
### What does this PR do?

Removes pluginId mapping

### Checklist
- [X] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: cordova-lib/spec-plugman/install.spec.js [hunk 1]
```diff
@@ -221,7 +221,7 @@ describe('install', function() {
             });
         });
 
-        it('Test 006 : should call fetch and convert oldID to newID', function(done) {
+        xit('Test 006 : should call fetch and convert oldID to newID', function(done) {
```
Comment (audreyso) [inline cordova-lib/spec-plugman/install.spec.js]: I don't think this test is needed any more??

Comment (stevengill) [inline cordova-lib/spec-plugman/install.spec.js]: yup

File: cordova-lib/src/plugman/uninstall.js [hunk 2]
```diff
@@ -165,14 +165,7 @@ module.exports.uninstallPlugin = function(id, plugins_dir, options) {
         deps.forEach(function (d) {
             var parsedSpec = pluginSpec.parse(d.id);
             deps_path = path.join(plugin_dir, '..', parsedSpec.id);
```
Comment (stevengill) [inline cordova-lib/src/plugman/uninstall.js]: don't need deps_path variable or parsedSpec anymore i think.

File: cordova-lib/src/plugman/fetch.js [hunk 3]
```diff
@@ -342,13 +322,6 @@ function copyPlugin(pinfo, plugins_dir, link) {
...
     var altDest;
 
-    //check if alternative id already exists in plugins directory
-    if(pluginMapperotn[pinfo.id]) {
-        altDest = path.join(plugins_dir, pluginMapperotn[pinfo.id]);
-    } else if(pluginMappernto[pinfo.id]) {
-        altDest = path.join(plugins_dir, pluginMappernto[pinfo.id]);
-    }
-
     if(fs.existsSync(altDest)) {
```
Comment (stevengill) [inline cordova-lib/src/plugman/fetch.js]: I think altDest doesn't need to exist anymore

File: cordova-lib/src/plugman/fetch.js [hunk 4]
```diff
@@ -257,7 +237,7 @@ function checkID(expectedIdAndVersion, pinfo) {
     var parsedSpec = pluginSpec.parse(expectedIdAndVersion);
 
     if (parsedSpec.id != pinfo.id) {
-        var alias = parsedSpec.scope ? null : pluginMappernto[parsedSpec.id] || pluginMapperotn[parsedSpec.id];
+        var alias = null;
         if (alias !== pinfo.id) {
```
Comment (stevengill) [inline cordova-lib/src/plugman/fetch.js]: delete this if + alias

Comment (audreyso) [inline cordova-lib/src/plugman/fetch.js]: kk thanks! @stevengill  If you remove this code, there are tests that go with it and they will fail so I removed those in the next commit. How does this look? https://github.com/apache/cordova-lib/pull/522/commits/a29720e6d6b2378d9e342a5b1bb46bef1db17fa8

Comment (stevengill) [inline cordova-lib/src/plugman/fetch.js]: LGTM

File: cordova-lib/src/plugman/fetch.js [hunk 5]
```diff
@@ -255,13 +235,7 @@ function checkID(expectedIdAndVersion, pinfo) {
     if (!expectedIdAndVersion) return;
 
     var parsedSpec = pluginSpec.parse(expectedIdAndVersion);
-
-    if (parsedSpec.id != pinfo.id) {
-        var alias = parsedSpec.scope ? null : pluginMappernto[parsedSpec.id] || pluginMapperotn[parsedSpec.id];
```
Comment (stevengill) [inline cordova-lib/src/plugman/fetch.js]: for this section, can you get rid of lines 260-263 but keep line 262.

It should look like 
```
 if (parsedSpec.id != pinfo.id) {
    throw new Error('Expected plugin to have ID "' + parsedSpec.id + '" but got "' + pinfo.id + '".');
}
```

That way, you can also undo your changes to cordova-lib/spec-plugman/fetch.spec.js. We should keep those tests I think.

Comment (stevengill) [inline cordova-lib/src/plugman/fetch.js]: I know this is the opposite of what I told you before. My bad 😛

Comment (audreyso) [inline cordova-lib/src/plugman/fetch.js]: Made some updates! Let me know if there is anything else! Thank you! :)

**Pull request closed without merging**